### PR TITLE
Bugfix: fix black frame between videos

### PR DESCRIPTION
**Pull request by emuraton**

### CODE REVIEW

File: src/SourceNodes/sourcenode.js [hunk 1]
```diff
@@ -385,7 +385,6 @@ class SourceNode extends GraphNode {
         }
 
         if (currentTime >= this._stopTime) {
-            clearTexture(this._gl, this._texture);
```
Comment (munwaikong) [inline src/SourceNodes/sourcenode.js:388]: what's the idea behind removing this line?

Comment (emuraton) [reply src/SourceNodes/sourcenode.js:388]: Right now we have a black frame between each videos (aka sections with a video node). The cause of this, is this line clearing the texture.

So we have two choices from my understanding:
- First option, we have a flag in the sourcenode to know if its the last video node of our layer, and if it is we clear the texture.
From a code pov inside studio it means we need to loop on every section chains and if it the last video node we add this flag.

- Second option, we always remove this black frame

I think we can go for second option as a quick fix, then later think about a better way to do things

**Review by Sacharified — COMMENTED** [review]
Have you tested this with different combinations of transitions and transparent videos?

**Comment by munwaikong** [comment]
Yea we tested this yesterday and applying this change will cause problems. 

This can't be applied

**Comment by emuraton** [comment]
Right, another try, with probably a simple / silly approach.

Using @gsouquet idea to defer the black frame on the next tick.
I'm joining screenshot showing black frame disappeared from an user pov, you can also try be yourself on http://29753d45.ngrok.io

Before
Screenshot 2020-04-14 at 12 16 11

After
Screenshot 2020-04-14 at 12 16 55

File: src/SourceNodes/sourcenode.js [hunk 2]
```diff
@@ -360,10 +360,14 @@ class SourceNode extends GraphNode {
         //update the current time
         this._currentTime = currentTime;
 
+        if(this._state === STATE.ended ){
+            clearTexture(this._gl, this._texture);
```
Comment (munwaikong) [inline src/SourceNodes/sourcenode.js:364]: we don't really want to be calling clearTexture for every RAF() - waste of resources?

File: src/SourceNodes/sourcenode.js [hunk 3]
```diff
@@ -384,8 +388,7 @@ class SourceNode extends GraphNode {
             this._state = STATE.playing;
         }
 
-        if (currentTime >= this._stopTime) {
-            clearTexture(this._gl, this._texture);
+        if (currentTime > this._stopTime) {
```
Comment (munwaikong) [inline src/SourceNodes/sourcenode.js]: is there a reason why you've changed from `>=` to `>` only?

File: src/SourceNodes/sourcenode.js (hunk 2 above)
Comment (emuraton) [reply src/SourceNodes/sourcenode.js:364]: there is a defensive if inside `clearTexture()`, it should protect us from this

File: src/SourceNodes/sourcenode.js (hunk 3 above)
Comment (emuraton) [reply src/SourceNodes/sourcenode.js]: good catch, I actually forgot to put it back

**Pull request closed without merging**